forc-pkg/src/manifest.rs

```diff
@@ -13,6 +13,26 @@ use sway_core::{language::parsed::TreeType, parse};
 pub use sway_types::ConfigTimeConstant;
 use sway_utils::constants;
 
+pub enum ManifestFile {
+    Package(Box<PackageManifestFile>),
+    Workspace(WorkspaceManifestFile),
+}
+
+impl ManifestFile {
+    /// Returns a `PackageManifestFile` if the path is within a package directory, otherwise
+    /// returns a `WorkspaceManifestFile` if within a workspace directory.
+    pub fn from_path(dir: &Path) -> Result<Self> {
+        let path = dir.to_path_buf().join("Forc.toml");
```

Comment:
Hmm I did not do that like that because from dir searches parent directories as well. Imagining a package residing in a workspace, If that package does not have `Forc.toml`, `from_dir` can find the workspace manifest. Let me try that in case I am missing something

Reply:
Yeah it seems like I was missing something 😄 I changed this place to mirror `{Package/Workspace}ManifestFile` `from_file` and `from_dir`